daemon/src/source_pair_import.rs

```diff
@@ -95,6 +96,9 @@ impl<'a> SourcePairImport<'a> {
         self.importer = importers.get_by_path(&self.source);
         Ok(self.importer.is_some())
     }
+    pub fn with_importer_contexts(&mut self, importer_contexts: &'a Vec<Box<dyn ImporterContext>>) {
```

Comment:
Those are not builder-style `with` methods, as those don't return `Self` so don't chain. I would name that `set_importer_contexts`. Same goes to `with_source_hash` and `with_meta_hash`.